openhtf/plugs/uart.py

```diff
@@ -0,0 +1,295 @@
+# Copyright 2014 Google Inc. All Rights Reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""SerialPlug used for serial communication.
+
+Requires the following keys in .yaml configuration file.
+Uses the openhtf.conf utility to access configuration file keys.
+
+.yaml file
+  SerialPlug:
+    serial_port: /dev/ttyUSB*     # serial dev name search format.
+    serial_baudrate: 19600
+    serial_parity: N              # PARITY_NONE, PARITY_EVEN, PARITY_ODD,
+                                    PARITY_MARK,
+                                    PARITY_SPACE = 'N', 'E', 'O', 'M', 'S'
+    serial_stopbits: 1            # STOPBITS_ONE, STOPBITS_ONE_POINT_FIVE,
+                                    STOPBITS_TWO = (1, 1.5, 2)
+    serial_bytesize: 8            # FIVEBITS, SIXBITS, SEVENBITS,
+                                    EIGHTBITS = (5, 6, 7, 8)
+    serial_rtscts: 0              # boolean type
+    serial_timeout: None
```

Comment:
Thinking a bit more about it, if these are config keys only, then a given test script it limited to a single serial device it can talk to. What happens if I want to talk to a DUT via serial as well as control a conveyor belt via serial? Approaches I can think of might include having OpenPort be able to take args to specify the serial settings then return some kind of connection object, or just letting test authors decorate the phase multiple times and use each instance however they want.